### PR TITLE
[Hotfix][Zeta] Fix submit job throw jobId NPE

### PR DESCRIPTION
**Pull request by chaorongzhi**
## Purpose of this pull request
Fix the problem that jobId is null and job submission fails, and add documentation comments.

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by lightzhao — APPROVED** [review]

**Comment by liugddx** [comment]
Can you add a unit test?

**Comment by chaorongzhi** [comment]
> Can you add a unit test?
Sure, I'll add unit tests to the e2e module.

**Comment by chaorongzhi** [comment]
> Can you add a unit test?

Sure, I'll add unit tests to the e2e module.

**Comment by chaorongzhi** [comment]
> Can you add a unit test?

@liugddx PTAL, I have added unit test.

File: docs/en/seatunnel-engine/rest-api.md [hunk 1]
```diff
@@ -192,8 +192,10 @@ network:
 > | jobId                | optional | string    | job id                            |
 > | jobName              | optional | string    | job name                          |
 > | isStartWithSavePoint | optional | string    | if job is started with save point |
-
-#### Body
+>
+> If the parameter isStartWithSavePoint is true, the jobId is required. If the parameter isStartWithSavePoint is false, the jobId is not required and will be generated by seatunnel
```
Comment (Hisoka-X) [inline docs/en/seatunnel-engine/rest-api.md]:
```suggestion
> If the parameter isStartWithSavePoint is true, the jobId is required. If the parameter isStartWithSavePoint is false, the jobId is not required and will be generated by SeaTunnel
```

**Comment by Hisoka-X** [comment]
cc @zhibinF maybe you should take a look.

**Pull request closed without merging**